ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Mempool.hs

```diff
@@ -296,134 +290,349 @@ theLedgerLens f x =
+  -> GenTx (ShelleyBlock proto era)
+  -> V.Validation (TxErrorSG era) ConwayMeasure
+txMeasureBabbage st tx@(ShelleyTx _txid tx') =
+    ConwayMeasure <$> txMeasureAlonzo st tx <*> refScriptBytes
+  where
+    utxo = SL.getUTxO $ tickedShelleyLedgerState st
+    txsz = SL.txNonDistinctRefScriptsSize utxo tx' :: Int
+
+    -- TODO recall the assertion in this module header that this logic should
+    -- be owned by Ledger, not by Consensus: Ledger prefers to delete these
+    -- checks after mainnet hard forks to Conway, rather than introduce the
+    -- necessary 'SL.PredicateFailure' constructors for Babbage (which would in
+    -- turn require a bump to "Ouroboros.Network.NodeToNode.Version" and
+    -- "Ouroboros.Network.NodeToClient.Version")
+    --
+    -- -- Babbage should have enforced a per-tx limit here, but did not.
+    -- refScriptBytes = pure $ ByteSize $ fromIntegral txsz
+    refScriptBytes :: V.Validation (TxErrorSG era) ByteSize
+    refScriptBytes =
+         (ByteSize $ fromIntegral txsz)
+         -- Reject it if it has more than 100 kibibytes of ref script.
+      <$ validateGuard
+           (err 1_000_000 (fromIntegral txsz) (fromIntegral limit))
+           (txsz <= limit)
+         -- Reject it if it has more than 50 kibibytes of ref script and does
+         -- not satisfy an additional fee of 'SL.tierRefScriptFee'.
+         --
+         -- TODO this additional fee check in particular doesn't truly belong
+         -- in 'txMeasure' (which only needs to enforce per-tx limits);
+         -- however, it's less confusing and less duplication to put it here
+         -- rather than
+         -- 'Ouroboros.Consensus.Shelley.Eras.defaultApplyShelleyBasedTx'
+      <* validateGuard
+           (err 100_000 (SL.unCoin fee) (SL.unCoin reqFee))
+           (reqFee <= fee)
+      where
+        limit = 100 * 1024
+
+        fee = tx' ^. L.bodyTxL . L.feeTxBodyL :: SL.Coin
+
+        reqFee :: SL.Coin
+        reqFee = if txsz <= 50 * 1024 then SL.zero else minFee SL.<+> addlFee
+
+        pparams = getPParams $ tickedShelleyLedgerState st
+        minFee  = L.getMinFeeTx pparams tx' 0
+        addlFee = SL.tierRefScriptFee 1.2 25600 15 txsz
+
+        -- As we are reusing an existing error message, we add a large number
+        -- to ensure users running into this are productively irritated and
+        -- post this error message somewhere where they can receive
+        -- help/context.
+        err :: Integer -> Integer -> Integer -> SL.ApplyTxError era
+        err shift l r = maxTxSizeUTxO (l + shift) (r + shift)
+
+-- | We anachronistically use 'ConwayMeasure' in Babbage.
+instance ( ShelleyCompatible p (BabbageEra c)
+         ) => TxLimits (ShelleyBlock p (BabbageEra c)) where
+
+  type TxMeasure (ShelleyBlock p (BabbageEra c)) = ConwayMeasure
+  txMeasure              _cfg st tx = runValidation $ txMeasureBabbage st tx
+  blockCapacityTxMeasure _cfg       = blockCapacityConwayMeasure
```

Comment:
This switches the 2.5mebi limit to 1mebi (... right?). Should be fine --- no legit block has needed that much.

Reply:
FTR this reduction for Babbage has already happened in #1168